izumi-reflect/izumi-reflect/src/main/scala-3/izumi/reflect/dottyreflection/ReflectionUtil.scala

```diff
@@ -37,16 +37,27 @@ private[dottyreflection] trait ReflectionUtil { this: InspectorBase =>
 
   extension (typeRef: TypeRef | ParamRef) {
     protected def _underlying: TypeRepr = {
-//      val underlying = typeRef
-//        .getClass.getMethods.collect { case m if m.getName == "underlying" => m }.head.invoke(
-//          typeRef,
-//          qctx.getClass.getMethods.collect { case m if m.getName == "ctx" => m }.head.invoke(qctx)
-//        )
-//      underlying.asInstanceOf[TypeRepr]
-
       // This works as a substitution for `TypeRef#underlying` call,
       // but I'm not sure if it's a reliable substitution.
-      typeRef.typeSymbol.owner._typeRef.memberType(typeRef.typeSymbol)
+
+//      typeRef.typeSymbol.owner._typeRef.memberType(typeRef.typeSymbol)
+
+      // No, It's not a reliable substitution. When used on a TypeParamRef it returns Any instead of the underlying TypeBounds
+      // https://github.com/lampepfl/dotty/issues/15799
+
+      val underlying = typeRef
```

Comment:
@neko-kai Another way to experiment using the compiler internals could be

```scala
import scala.reflect.Selectable.reflectiveSelectable
type Context
type InternalQuotes = {
  def ctx: Context
}
type InternalTypeRef = {
  def underlying(using Context): TypeRepr
}
given Context = qctx.asInstanceOf[InternalQuotes].ctx
typeRef.asInstanceOf[InternalTypeRef].underlying

Reply:
@nicolasstucki I'll keep this in mind. Thanks! 👍